prettier/entrypoint.sh

```diff
@@ -258,6 +251,14 @@ else
   else
     git commit -n -m "$INPUT_COMMIT_MESSAGE" ${INPUT_COMMIT_OPTIONS:+"$INPUT_COMMIT_OPTIONS"} || commit_canary=
   fi
+
+  # case when --write is used with dry-run so if something is unpretty there will always have _git_changed
+  if _dry_run; then
+    summarize_changes 'Unpretty Files Changes'
+    echo "Finishing dry-run. Exiting before committing." | tee -a "$GITHUB_STEP_SUMMARY"
+    exit 1
+  fi
+
```

Comment:
Since we have two `git push` in this workflow, I think it makes sense to extract the dry-run check into a function and call it immediately before each `git push`.